test.py

```diff
@@ -0,0 +1,59 @@
+import time
+import os
+import sys
+import pytest
+import subprocess
+
+# Add the project root directory to the system path
+root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+sys.path.append(root_dir)
+
+from client.client import Client
+
+TEST_PPT_PATH = "/Users/damiantissembaum/Documents/year_3/TEMP/Tests-3.pptx"
+API_PROCESS = None
+
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_teardown_system():
+    # Start up the API and explainer
+    api_process_1 = subprocess.Popen(['python', 'flaskServer/main.py'])
+    api_process_2 = subprocess.Popen(['python', 'explainer/Explainer.py'])
+
+    time.sleep(5)  # or however long they need to be ready
+    yield
+
+    if api_process_1 is not None:
+        api_process_1.terminate()
+        api_process_1 = None
+        print("API process stopped")
+
+    if api_process_2 is not None:
+        api_process_2.terminate()
+        api_process_2 = None
+        print("explainer process stopped")
+
+
+def test_end_to_end():
+    # Step 1: Create client
```

Comment:
great that you're explaining the stages, should also document the function itself.